src/Core/Services/SendService.cs

```diff
@@ -0,0 +1,275 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Bit.Core.Abstractions;
+using Bit.Core.Enums;
+using Bit.Core.Models.Data;
+using Bit.Core.Models.Domain;
+using Bit.Core.Models.Request;
+using Bit.Core.Models.Response;
+using Bit.Core.Models.View;
+using Newtonsoft.Json;
+
+namespace Bit.Core.Services
+{
+    public class SendService : ISendService
+    {
+        public static string SEND_KEY(string userId) => string.Format("sends_{0}", userId);
+        private List<SendView> _decryptedSendsCache;
+        private readonly ICryptoService _cryptoService;
+        private readonly IUserService _userService;
+        private readonly IApiService _apiService;
+        private readonly IStorageService _storageService;
+        private readonly II18nService _i18nService;
+        private readonly ICryptoFunctionService _cryptoFunctionService;
+        private Task<List<SendView>> _getAllDecryptedTask;
+
+
+        public SendService(
+            ICryptoService cryptoService,
+            IUserService userService,
+            IApiService apiService,
+            IStorageService storageService,
+            II18nService i18nService,
+            ICryptoFunctionService cryptoFunctionService)
+        {
+            _cryptoService = cryptoService;
+            _userService = userService;
+            _apiService = apiService;
+            _storageService = storageService;
+            _i18nService = i18nService;
+            _cryptoFunctionService = cryptoFunctionService;
+        }
+
+        public async Task ClearAsync(string userId)
+        {
+            await _storageService.RemoveAsync(SEND_KEY(userId));
+            ClearCache();
+        }
+
+        public void ClearCache() => _decryptedSendsCache = null;
+
+        public async Task DeleteAsync(params string[] ids)
+        {
+            var userId = await _userService.GetUserIdAsync();
+            var sends = await _storageService.GetAsync<Dictionary<string, SendData>>(SEND_KEY(userId));
+
+            if (sends == null)
+            {
+                return;
+            }
+
+            foreach (var id in ids)
+            {
+                sends.Remove(id);
+            }
+
+            await _storageService.SaveAsync(SEND_KEY(userId), sends);
+            ClearCache();
+        }
+
+        public async Task DeleteWithServerAsync(string id)
+        {
+            await _apiService.DeleteSendAsync(id);
+            await DeleteAsync(id);
+        }
+
+        public async Task<(Send send, CipherString encryptedFileData)> EncryptAsync(SendView model, byte[] fileData,
+            string password, SymmetricCryptoKey key = null)
+        {
+            var send = new Send();
+            CipherString encryptedFileData = null;
```

Comment:
encryptedFileData var + assignment can be moved down, and the `new Send { ... }` can be simplified into a simple object constructor format.